src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -100,6 +107,20 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             }
         }
 
+        IDataEncoder GetDataEncoder(TransferMode requestedTransferMode, TransferMode actualTransferMode)
+        {
+            if (requestedTransferMode == TransferMode.Binary && actualTransferMode == TransferMode.Text)
+            {
+                // This is for instance for SSE which is a Text protocol and the user wants to use a binary
+                // protocol so we need to encode messages.
+                return new Base64Encoder();
```

Comment:
They implement an interface so they can either be singletons or some code changes will be required